api/ai/new/multiple_round_robin_with_adjusted_winner/multiple_round_robin.py

```diff
@@ -0,0 +1,204 @@
+"""
+N = list of projects
+H = list of students
+V = value function
+
+Find biased allocation using MRR
+while H is not empty:
+    Order N by team value V(n) ascending (each project has 1 team)
+    if ∃ student s ∈ H, project n ∈ N that u_n(s) > 0:
+        add k dummy students h_k to H, that len(H) + k = a * len(N) with a is an arbitrary constant
+
+    for n in N:
+        if not ∃ project n ∈ N that V(n) > 0 and there is h_k ∈ H:
+            h_k -> n.team
+        else:
+            student_max = argmax student s ∈ H of V(n with s)
+            student_max -> n.team
+            update V(n)
+            Reorder N by team value V(n) ascending (each project has 1 team)
+
+Apply Adjusted Winner Algorithm to the found allocation
+for team_i in N:
+    for team_j in N:
+        if team_i is not OEF1 w.r.t team_j:
+            for student in team_j:
+                if (team_i + student).value > team_i.value:
+                    Assign student to team_i
+
+"""
+from heapq import heappush, heappop
+from typing import List, Dict
+
+from api.ai.new.interfaces.algorithm import Algorithm
+from api.ai.new.interfaces.algorithm_options import MultipleRoundRobinAlgorithmOptions
+from api.ai.new.interfaces.team_generation_options import TeamGenerationOptions
+from api.ai.new.multiple_round_robin_with_adjusted_winner.custom_models import (
+    TeamWithValues,
+    StudentProjectValue,
+)
+from api.ai.new.multiple_round_robin_with_adjusted_winner.utils import (
+    is_oef1,
+    calculate_value,
+)
+from api.models.project import Project
+from api.models.student import Student
+from api.models.team_set import TeamSet
+
+
+class MultipleRoundRobinWithAdjustedWinnerAlgorithm(Algorithm):
+    def __init__(
+        self,
+        algorithm_options: MultipleRoundRobinAlgorithmOptions,
+        team_generation_options: TeamGenerationOptions,
+        projects: List[Project],
+    ):
+        super().__init__(algorithm_options, team_generation_options)
+        self.projects = projects
+
+    @staticmethod
+    def _get_values_heap(
+        students: List[Student], teams: List[TeamWithValues]
+    ) -> Dict[int, List[StudentProjectValue]]:
+        """
+        Calculate each student values to each project and push them into a heap
+
+        This run in O(N * H)
+        """
+        # Each project will have its own heap, showing how effective each student to that project
+        values_heap: Dict[int, List[StudentProjectValue]] = {
+            team.project.id: [] for team in teams
+        }
+        for team in teams:
+            project = team.project
+            for student in students:
+                new_value = StudentProjectValue(student, project)
+                heappush(values_heap[project.id], new_value)
+
+        return values_heap
+
+    @staticmethod
```

Comment:
why make these methods static if you're just calling them with `self.` anyway?

If they are truly non-dependant on the `self.` then why aren't they utils?

(note: not every warning from pycharm means you need to fix it )